src/java/picard/analysis/directed/TargetMetricsCollector.java

```diff
@@ -425,32 +585,47 @@ private void calculateTargetCoverageMetrics() {
             int zeroCoverageTargets = 0;
             int depthIndex = 0;
             double totalCoverage = 0;
-            int basesConsidered = 0;
+            long numTargetBasesAtZeroCoverage = 0; // should equal (targetBasesDepth[1] - targetBasesDepth[0]) below
+            final Histogram<Integer> targetCoverageHistogram = new Histogram<Integer>();
```

Comment:
I feel like this is redundant with depths[] above.  Could you just move the Arrays.sort(depths) above the line where you compute median coverage, and then get the median from depths[]?